haas/api.py

```diff
@@ -258,34 +275,57 @@ def node_connect_network(node, nic, network):
 
     project = node.project
 
+    allocator = get_network_allocator()
+    if channel is None:
+        channel = allocator.get_default_channel(db)
+
```

Comment:
More of a stylistic comment (take it or leave it), but it might make sense to move this closer to where the 2 other channel-related error checks are so that the logic dealing with 'channel' is close together and easy to see.